src/Fetcher/TululaFetcher.php

```diff
@@ -0,0 +1,241 @@
+                }
+            }
+
+            yield $this->getUrl() => $this->denormalizeConferences($sameTagConferences, $tag);
+        }
+    }
+
+    public function denormalizeConferences(array $rawConferences, Tag $tag): array
+    {
+        $conferences = [];
+        foreach ($rawConferences as $rawConference) {
+            if (!$rawConference['isOnline']) {
+                // There are cases where Tulula doesn't provide city or state so the string might be empty
+                if (!empty($city = $rawConference['venue']['city'] ?: $rawConference['venue']['state'])) {
+                    $query = sprintf('%s', $city);
+                    $continent = $this->locationGuesser->getContinent($query);
+
+                    if (!$continent instanceof Continent || !$continent->getEnabled()) {
+                        continue;
+                    }
+                }
+            }
+
+            $startDate = \DateTimeImmutable::createFromFormat('Y-m-d', $rawConference['dateStart']);
+
+            // In case of invalid startDate, we skip the conference. It will be handled again later.
+            if (!$startDate) {
+                continue;
+            }
+
+            $conference = new Conference();
+            $conference->setSource(self::SOURCE);
+            $conference->setSlug($rawConference['slug']);
+            $conference->setName($rawConference['name']);
+            $conference->setStartAt($startDate);
+            $conference->setSiteUrl($rawConference['url']);
+            $conference->addTag($tag);
+
+            if ($rawConference['isOnline']) {
+                $conference->setCity('Online');
+                $conference->setCountry('Online');
+            } else {
+                $conference->setCity($rawConference['venue']['city'] ?: $rawConference['venue']['state'] ?: null);
+                $conference->setCountry($rawConference['venue']['countryCode']);
+            }
+
+            $excluded = false;
+            foreach ($this->excludedTags as $excludedTag) {
+                if (fnmatch($excludedTag->getName(), $rawConference['name'], FNM_CASEFOLD)) {
+                    $excluded = true;
+                    break;
+                }
+            }
+            $conference->setExcluded($excluded);
+            if ($rawConference['dateEnd']) {
+                $endDate = \DateTimeImmutable::createFromFormat('Y-m-d', $rawConference['dateEnd']);
+                $conference->setEndAt($endDate);
+            }
+
+            if ($rawConference['cfpUrl']) {
+                $conference->setCfpUrl($rawConference['cfpUrl']);
+            }
+
+            if ($rawConference['cfpDateEnd']) {
+                $cfpEndAt = \DateTimeImmutable::createFromFormat('Y-m-d', $rawConference['cfpDateEnd']);
+                $conference->setCfpEndAt($cfpEndAt);
+            }
+            $conferences[] = $conference;
+        }
+
+        return $conferences;
+    }
+
+    private function queryTululaApi()
```

Comment:
1. why no typehint ?
2. The naming seems bad to me. You don't get an "API", but some events